src/gridfs/download.rs

```diff
@@ -0,0 +1,181 @@
+use tokio::io::AsyncWriteExt;
+use tokio_util::compat::FuturesAsyncWriteCompatExt;
+
+use super::{options::GridFsDownloadByNameOptions, FilesCollectionDocument, GridFsBucket};
+use crate::{
+    bson::{doc, Bson},
+    error::{ErrorKind, Result},
+    options::{FindOneOptions, FindOptions},
+};
+
+impl GridFsBucket {
+    /// Downloads the contents of the stored file specified by `id` and writes
+    /// the contents to the `destination`.
+    pub async fn download_to_tokio_writer<T>(&self, id: Bson, destination: &mut T) -> Result<()>
+    where
+        T: tokio::io::AsyncWrite + std::marker::Unpin,
+    {
+        let options = FindOneOptions::builder()
+            .read_concern(self.read_concern().cloned())
+            .selection_criteria(self.selection_criteria().cloned())
+            .build();
+
+        let file = match self
+            .inner
+            .files
+            .find_one(doc! { "_id": &id }, options)
+            .await?
+        {
+            Some(fcd) => fcd,
+            None => {
+                return Err(ErrorKind::InvalidArgument {
+                    message: format!("couldn't find file with id {}", &id),
+                }
+                .into());
+            }
+        };
+        self.download_to_tokio_writer_common(file, destination)
+            .await
+    }
+
+    /// Downloads the contents of the stored file specified by `id` and writes
+    /// the contents to the `destination`.
```

Comment:
The method name implies this of course, but can we explicitly mention that the destination here implements the `AsyncWrite` trait from the `futures 0.3` crate, and link to it? Ditto for the tokio ones. Might also be worth mentioning that once `std` stabilizes `AsyncWrite`, this method will be deprecated (ditto for tokio ones on this too).

Reply:
Since the `to_futures_0_3` methods are just using `compat_write` under the hood and are temporary, what would people think about leaving them out entirely and recommending users do this directly if needed in documentation?

Reply:
@abr-egn I'd be fine with that

Reply:
It's a little inconsistent with our various cursor APIs that implement the `futures` version of `Stream`, but having seen what the implementation actually entails here, I think I agree that maybe these separate methods probably aren't worth it. Instead, if we want to have consistency, maybe we should just also implement tokio's `Stream` on our cursor types. Then we can omit the `futures` one here.

Reply:
Would it make more sense to have only the `futures` implementations then? That way we're runtime-agnostic and in line with what we already support for `Cursor`. The tokio compat utils go both ways, so I think the user conversion code would be basically the same regardless of which library we choose to support in our API.